src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -700,15 +721,22 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
             return;
         }
 
+        const auto shellType = _ParsePreferredShellType(settingsResponse);
+        if (!shellType.has_value())
+        {
+            _WriteStringWithNewline(RS_(L"AzureInvalidUserSettings"));
+            _transitionToState(ConnectionState::Failed);

```

Comment:
If they're missing a preferred shell type, is there a way to just fall back to something reasonable, instead of failing entirely? Maybe we could display the warning, but then use PowerShell, and if they want `bash` then at least they'll know why powershell was chosen
/cc @DHowett

Reply:
@zadjii-msft - absolutely. Actually, I already implemented something similar for: https://github.com/microsoft/terminal/issues/4987

In that ticket I allow the user to always select a shell, where the preferred one is marked as default.
But I guess that ticket will be stuck on product, because UX is not trivial (extra user input is introduced).

So I can probably adjust the logic from there to here. I.e., read the preferred shell.. if no preferred shell was found - allow the user to select the shell manually.

I will do the adjustment, but only if you help me push it 😊